### ISSUE TITLE
Truncating table trim methodology issues

### ISSUE THREAD

**Issue by kubouch** [nushell/nushell#6023]
### Describe the bug

The truncating table trim mode seems to favor columns that do not need much truncating. I found several issues:
1. The index column on the left is truncated immediately. I think it should either not truncate that column at all, or truncate it with the least priority.
2. From the table below, it can be seen that most of the horizontal space is occupied by the `target` column. It would be enough to just trim that one column instead of trimming every column by a little. Maybe the decision algorithm could prioritize trimming wider columns first?

### How to reproduce

```
> ls -al
 ..         name          type                               target                                mode      num_lin..     size        modified
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 ..   .BitwigStudio      dir                                                                     rwxr-xr..          10     4.0 KiB   3 weeks ag..
 ..   .Xauthority        file                                                                    rw-----..           1        49 B   a month ag..
 ..   .actrc             file                                                                    rw-r--r..           1       164 B   4 months a..
 ..   .android           dir                                                                     rwxr-xr..           4     4.0 KiB   5 months a..
 ..   .auto_base16.n..   symli..   /home/kubouch/git/forks/nu_scripts/engine-q/base16/auto_b..   rwxrwxr..           1        65 B   6 months a..
 ..   .bash_history      file                                                                    rw-----..           1   11.7 Ki..   20 hours a..
 ..   .bash_logout       file                                                                    rw-r--r..           1        21 B   7 months a..
 ..   .bash_profile      symli..   /home/kubouch/git/dotfiles/bash/bash_profile                  rwxrwxr..           1        44 B   7 months a..
 ..   .bashrc            symli..   /home/kubouch/git/dotfiles/bash/bashrc                        rwxrwxr..           1        38 B   7 months a..
 ..   .cache             dir                                                                     rwxr-xr..          37     4.0 KiB   4 days ago
 ..   .cargo             dir                                                                     rwxr-xr..           6     4.0 KiB   19 hours a..
 ..   .conda             dir                                                                     rwxr-xr..           2     4.0 KiB   6 months a..
 ..   .config            dir                                                                     rwxr-xr..          45     4.0 KiB   4 days ago
 ..   .electron-gyp      dir                                                                     rwxr-xr..           3     4.0 KiB   3 months a..
 ..   .emacs.d           dir                                                                     rwx----..           3     4.0 KiB   a month ag..
 ..   .fehbg.nu          symli..   /home/kubouch/git/dotfiles/feh/fehbg.nu                       rwxrwxr..           1        39 B   4 months a..
(...)
```

### Expected behavior

I would expect the indices to either not be truncated, and the `target` column to be truncated more heavily.

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.65.1                                          |
| branch             |                                                 |
| commit_hash        | f3036b8cfd4dd6177125fb1e1d1898009097734e        |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.62.0 (a8314ef7d 2022-06-27)             |
| rust_channel       | stable-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.62.0 (a748cf5a3 2022-06-08)             |
| pkg_version        | 0.65.1                                          |
| build_time         | 2022-07-11 23:59:41 +03:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  | gstat                                           |

### Additional context

```
> $env.config.table_trim

────────────────────────────────
 methodology         truncating
 truncating_suffix   ..
```

**Comment by fdncred** [comment]
i really dislike entire columns being `......`. i really always wan to see some text to get me a hint of what that column is supposed to represent. I guess i don't really want to see any values anywhere that are entirely dots. whether that be headers, footers, or content.  just my opinion.

**Comment by kubouch** [comment]
Yeah, looks better to me but I agree with @fdncred that having values as dots only is not ideal.

I just managed to get this table :laughing: 
```
> ls -l
 ..   .   .   .   .   .   .
────────────────────────────
 ..   .   .       .   .   .
 ..   .   .       .   .   .
 ..   .   .       .   .   .
 ..   .   .       .   .   .
 ..   .   .       .   .   .
```
vs. (same window size)
```
  #           name            type     size        modified
──────────────────────────────────────────────────────────────
  0   CODE_OF_CONDUCT.md      file     3.4 KiB   7 months ago
  1   CONTRIBUTING.md         file     1.6 KiB   a month ago
  2   Cargo.lock              file   136.3 KiB   an hour ago
  3   Cargo.toml              file     3.7 KiB   2 hours ago
  4   LICENSE                 file     1.1 KiB   3 months ago
```
You can see it shortened more than necessary, there is still a lot of space left. In that full-dots example, we could make the decision to omit columns with `...` earlier and display at least something. Also, the `..` in the index column could also be numbers without increasing the horizontal space (the number of items was 20 -- two digits).

**Comment by fdncred** [comment]
i just noticed something that may be related with table width being the wrong size https://github.com/nushell/nushell/pull/6028#issuecomment-1182257616

**Comment by zhiburt** [comment]
> i really dislike entire columns being ....... i really always wan to see some text to get me a hint of what that column is supposed to represent. I guess i don't really want to see any values anywhere that are entirely dots. whether that be headers, footers, or content. just my opinion.

That's what you get when you have a bigger suffix then the available space.
And we discussed here.

https://github.com/nushell/nushell/pull/5999#issuecomment-1181805372

>  i really dislike entire columns being ....... i really always wan to see some text to get me a hint of what that column is supposed to represent. 

Do you propose to not apply truncating to the headers?

> I just managed to get this table laughing

Could you @kubouch show how you get it?
It's indeed look strange from first look.

**Comment by fdncred** [comment]
> That's what you get when you have a bigger suffix then the available space.
> And we discussed here.

Oh, i see what you mean now. Sorry, I didn't understand what you meant. I wonder how we can prevent having that big of a suffix?

> Do you propose to not apply truncating to the headers?

I'm fine with truncation of headers but not complete truncation where the header is .......  I wonder what @kubouch thinks about it?

**Comment by zhiburt** [comment]
> Oh, i see what you mean now. Sorry, I didn't understand what you meant. I wonder how we can prevent having that big of a suffix?

It's user defined so not a lot we can do I think.

> Do you propose to not apply truncating to the headers?

This is how it would look like

![image](https://user-images.githubusercontent.com/20165848/178601603-9214d46d-3f49-40cb-a98f-6da50683dbd5.png)

**Comment by kubouch** [comment]
@zhiburt I just ran `ls -l` in a sufficiently small window. I made the window just wide enough to fill `ls` without truncating, then ran `ls -l` in the same window.

@fdncred I would also discourage truncating the headers. On the other hand, if the header is super-long, then truncating it might be a good idea.

I feel like these issues will need to be polished over time. It might take some time and experimentation to figure out the heuristics that feel right.
